src/base/memory_monitor_sr.cpp

```diff
@@ -674,7 +674,7 @@ void mmon_aggregate_server_info (MMON_SERVER_INFO &info)
 void mmon_aggregate_module_info (int module_index, std::vector<MMON_MODULE_INFO> &info)
 {
   assert (mmon_Gl != nullptr);
-  assert (module_index < (int)MMON_MODULE_LAST && module_index >= 0);
+  assert (module_index <= (int)MMON_MODULE_LAST && module_index >= 0);
```

Comment:
Why is `module_index == MMON_MODULE_LSA` allowed? Could you describe the case?

Reply:
Because client send module index start to 1(except all) but MMON_MODULE_ID start to 0. This occurs due to the design intention of avoiding the creation of a dummy. To align this discrepancy in indexes, 1 is subtracted from the module_index received from the client in memory_monitor::aggregate_module_info().

Reply:
Ok, I understand. Hmm... to make the rule clearer, how about putting the subtraction in the client-side? I mean to have the server always assume that the module_index starts from 0.